### ISSUE TITLE
static async member

### ISSUE THREAD

**Issue by shaunc** [eslint/eslint#8597]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 7.10.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Using babel, but eslinting on source.

**Please show your full configuration:**

```
// eslintrc.js
module.exports = {
  root: true,
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module'
  },
  extends: 'eslint:recommended',
  env: {
    node: true,
    es6: true
  },
  rules: { }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
class Foo {
  static async foo() {
    await Promise.resolve(1);
  }
}
```

**What did you expect to happen?**

(no error)

**What actually happened? Please include the actual, raw output from ESLint.**
```
/Volumes/Macintosh_HD/Users/shauncutts/src/gabbage/src/foo.js
  3:16  error  Parsing error: Unexpected token foo

✖ 1 problem (1 error, 0 warnings)
```
**Update**: `export async function....` seems to have a problem as well. Am I missing a config switch?

**Comment by ljharb** [comment]
`ecmaVersion` 6 doesn't have `async function` - it was added in ES2018. Set `ecmaVersion` to `2018`.

**Comment by ljharb** [comment]
oops, i mean ES 2017. will edit and delete this correction.

**Comment by platinumazure** [comment]
As @ljharb noted, if you want to use async functions, you need to use `ecmaVersion: 2017`.

**Comment by shaunc** [comment]
Ah -- thought I must be missing something. Hmm... its not even a documentation bug. :)
